### PR TITLE
Created Location Details View

### PR DESCRIPTION
**Pull request by peytonshetler**
## Summary
So, I'm not sure how I want to organize the information that'll be coming from Firebase so, for now, I just created top and bottom containers to hold future UI elements (Also, a basic label at the top just as a visual reference).  

#### Characteristics

- [X] Feature completed
- [X] Experimental
- [X] UI affected (screen shots included)

### CODE REVIEW

File: Been-There/Been-There/Controller/LocationDetailsVC.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//  LocationDetailsViewController.swift
+//  Been-There
+//
+//  Created by Peyton Shetler on 4/2/19.
+//  Copyright © 2019 Peyton Shetler. All rights reserved.
+//
+// Make a basic view to show details
+import UIKit
+
+class LocationDetailsViewController: UIViewController {
```
Comment (JZDesign) [inline Been-There/Been-There/Controller/LocationDetailsVC.swift]: I noticed that the class name doesn't match the file name. There is no right way between the two, it's personal choice, so whatever you choose to do is fine as long as you're consistent. 

If this is how you want to name your files moving forward, for your next PR go ahead and change all of the file names to match accordingly as `VC` instead of `ViewController`. Otherwise rename this file to end with `ViewController` so you don't have a bunch of naming conventions. 

There is a widely adopted design pattern called `Domain Driven Design` where the word domain really refers to the language you use and what it refers to in your code. You'll often hear of a `DSL` or `Domain Specific Language` that spells out what means what in programming. This relates pretty closely. If you don't choose a standard and stick to it, it could confuse the recruiters and developers you send to this repo.

File: Been-There/Been-There/Controller/UI+LocationDetails.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        exampleTitle.heightAnchor.constraint(equalToConstant: 40).isActive = true
+        exampleTitle.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        exampleTitle.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+        exampleTitle.text = "Location Details Screen!"
+        exampleTitle.textAlignment = .center
+        exampleTitle.font = exampleTitle.font.withSize(28)
+        topContainer.addSubview(exampleTitle)
+    }
+    
+    func configureContainer(_ container: UIView, _ completion: @escaping (_ newContainer: UIView) -> Void) {
```
Comment (JZDesign) [inline Been-There/Been-There/Controller/UI+LocationDetails.swift]: I _think_ saw a similar function in other files when I reviewed some of your other PRs. If so, this may be a good function to have in a `UIViewController` extension file so it's available to all of your future View Controllers.

File: Been-There/Been-There/Client/ViewController/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//  ViewController.swift
+//  Been-There
+//
+//  Created by Peyton Shetler on 4/3/19.
+//  Copyright © 2019 Peyton Shetler. All rights reserved.
+//
+
+import UIKit
+
+public extension UIViewController {
```
Comment (JZDesign) [inline Been-There/Been-There/Client/ViewController/ViewController.swift]: Awesome start! I think you could change this up a bit.

```

extension UIViewController {
    // Containers
    
    func configureChildContainer(_ child: UIView, inParent parentView: UIView, withHeightMultiplier multiplier: Double? = 1.0) {
        child.translatesAutoresizingMaskIntoConstraints = false
        child.widthAnchor.constraint(equalTo: parentView.widthAnchor).isActive = true
        child.centerXAnchor.constraint(equalTo: parentView.centerXAnchor).isActive = true
        parentView.addSubview(child)
        child.heightAnchor.constraint(equalTo: parentView.heightAnchor, multiplier: multiplier).isActive = true
        child.bottomAnchor.constraint(equalTo: parentView.bottomAnchor).isActive = true
    }
```

Then you can call it like 

```
    func setupBottomContainer() {
        configureChildContainer(bottomContainer, inParent: view, withHeightMultiplier: 3/4)
        bottomContainer.backgroundColor = .blue
        
    }
```

File: Been-There/Been-There/Controller/LocationDetailsVC.swift (hunk 1 above)
Comment (peytonshetler) [reply Been-There/Been-There/Controller/LocationDetailsVC.swift]: We chatted about this via Slack.  Very good to know.  I'm gonna stick with the non-shorthand way of naming in the future to promote clarity👍🏻

File: Been-There/Been-There/Controller/UI+HomeScreen.swift [hunk 4]
```diff
@@ -40,29 +40,15 @@ extension HomeScreenViewController {
         screenTitle.centerYAnchor.constraint(equalTo: topContainer.centerYAnchor).isActive = true
     }
     
-//    func setupTopContainer() {
-//        UIViewController.configureContainer(topContainer, view, view, view) { (topContainer) in
-//            topContainer.heightAnchor.constraint(equalTo: self.view.heightAnchor, multiplier: 1/8).isActive = true
-//            topContainer.topAnchor.constraint(equalTo: self.view.topAnchor).isActive = true
-//        }
-//    }
-
```
Comment (peytonshetler) [inline Been-There/Been-There/Controller/UI+HomeScreen.swift]: I deleted all the old setup functions in this commit as well.  I should've done a separate commit for that, apologies.

File: Been-There/Been-There/Client/ViewController/ConfigureChildContainerExtension.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    // Containers
+    class func configureChildContainer(_ child: UIView, inParent parentView: UIView, withHeightMultiplier multipier: CGFloat? = 1.0,_ completion: @escaping (_ sameChild: UIView) -> Void) {
+        child.translatesAutoresizingMaskIntoConstraints = false
+        parentView.addSubview(child)
+        child.widthAnchor.constraint(equalTo: parentView.widthAnchor).isActive = true
+        child.centerXAnchor.constraint(equalTo: parentView.centerXAnchor).isActive = true
+        child.heightAnchor.constraint(equalTo: parentView.heightAnchor, multiplier: multipier!).isActive = true
+        completion(child)
+    }
+}
```
Comment (peytonshetler) [inline Been-There/Been-There/Client/ViewController/ConfigureChildContainerExtension.swift]: This is the new container function!  Tried to use more English-sounding parameter labels like you did, haha.  The only thing that changes at this point in my containers is the Y position..some are Y-center, while some are placed using the topAnchor.  I was wondering if there's a savvy way to solve this within this func declaration, (maybe using a switch case?) or if thats unnecessarily complicated haha.

File: Been-There/Been-There/Client/ViewController/ConfigureChildContainerExtension.swift [hunk 6]
```diff
@@ -0,0 +1,21 @@
+//
```
Comment (JZDesign) [inline Been-There/Been-There/Client/ViewController/ConfigureChildContainerExtension.swift]: The file name here should match the structure as the others. If you're extending the file name should depict what is being extended and what the added functionality pertains to. This format is the standard approach `ViewController+ConfigureChildContainer.swift` or the inverse like you have in other files.

Comment (peytonshetler) [reply Been-There/Been-There/Client/ViewController/ConfigureChildContainerExtension.swift]: Ah, good catch!  It has been changed!
